gdshelpers/parts/waveguide.py

```diff
@@ -274,17 +239,26 @@ def find_y(y):
         else:
             sample_width = np.array([(width if width else self.current_port.width), ])
 
-        # Now we have everything to calculate the polygon
-        poly_path_1 = sample_coordinates + sample_width[:, None] / 2 * sample_coordinates_d1_normed_ortho
-        poly_path_2 = sample_coordinates - sample_width[:, None] / 2 * sample_coordinates_d1_normed_ortho
-
-        assert shapely.geometry.LineString(np.concatenate([poly_path_1, poly_path_2[::-1, :]])).is_simple, \
-            'Outer lines of parameterized wg intersect. Try using lower bend radii or smaller a smaller wg'
+        if sample_width.ndim == 1:
+            sample_width = sample_width[..., None]
```

Comment:
What happens if the provided width has different length (other than 1) than that of the current port?  Should this throw an error?

Reply:
Ah, I just realized that you just accept changing width numbers everywhere, so I guess it's ok...

Reply:
The width of the added part depends only on the parameter `width` and not on the width of the current port. Therefore we'd just see a discrete change of the width, but it should not be a problem here.
For add_straight_segment and add_bend it would be a problem if the lengths are different. But the question would be, how we could interpret a change of the widths correctly for linear tapering (except the case where 4n entries are added/removed to the width array, where we could assume that the missing 2n entries should be tapered to 0 on both sides).
But maybe for that case an error would be the more reasonable choice, as the user could define the behavious by changing the width of the waveguide before adding the segment?